File: app/src/main/java/com/kabouzeid/gramophone/views/basic/SwitchX.java
package com.kabouzeid.gramophone.views.basic;

import android.content.Context;
import android.util.AttributeSet;
import android.view.View;

import androidx.appcompat.widget.SwitchCompat;

import com.kabouzeid.appthemehelper.ATH;
import com.kabouzeid.appthemehelper.ThemeStore;

/**
 * @author Aidan Follestad (afollestad)
 */
public class SwitchX extends SwitchCompat {

    public SwitchX(Context context) {
        super(context);
        init(context, null);
    }

    public SwitchX(Context context, AttributeSet attrs) {
        super(context, attrs);
        init(context, attrs);
    }

    public SwitchX(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init(context, attrs);
    }

    private void init(Context context, AttributeSet attrs) {
        ATH.setTint(this, ThemeStore.accentColor(context));
    }

    @Override
    public boolean isShown() {
        return getParent() != null && getVisibility() == View.VISIBLE;
    }
}

File: app/src/main/java/lib/phonograph/view/SwitchX.kt
/*
 * Copyright (c) 2022 Abou Zeid (kabouzeid) (original author)
 */
package lib.phonograph.view

import android.content.Context
import android.util.AttributeSet
import androidx.appcompat.widget.SwitchCompat
import mt.pref.ThemeColor.accentColor
import mt.tint.viewtint.setTint
import mt.util.color.isWindowBackgroundDark

/**
 * @author Aidan Follestad (afollestad)
 */
class SwitchX : SwitchCompat {
    constructor(context: Context) : super(context) {
        init(context, null)
    }

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
        init(context, attrs)
    }

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {
        init(context, attrs)
    }

    private fun init(context: Context, attrs: AttributeSet?) {
        this.setTint(accentColor(context), isWindowBackgroundDark(context))
    }

    override fun isShown(): Boolean =
        parent != null && visibility == VISIBLE
}